lib/typed.js

```diff
@@ -337,6 +337,26 @@
         var ch, ch2;
 
         value = advance();
+
+        // check for import("./path/to/module").Foo type
+        var importStart = 'import(';
+        if (source.slice(index - 1, index - 1 + importStart.length) === importStart) {
+            value = importStart;
+            index += importStart.length - 1;
+            // scan until closing )
+            while (index < length && source.charAt(index) !== ')') {
```

Comment:
Is ')' a valid filesystem character? If so then you'll need to check for quotes, etc.

Note that Typescript parses *any* type inside the parentheses, but afterwards checks that it is a string literal.